packages/cubejs-teradata-jdbc-driver/src/TeradataDriver.ts

```diff
@@ -0,0 +1,267 @@
+  N: 'NUMBER',
+  PD: 'PERIOD(DATE)',
+  PM: 'PERIOD(TIMESTAMP WITH TIME ZONE)',
+  PS: 'PERIOD(TIMESTAMP)',
+  PT: 'PERIOD(TIME)',
+  PZ: 'PERIOD(TIME WITH TIME ZONE)',
+  SC: 'INTERVAL SECOND',
+  SZ: 'TIMESTAMP WITH TIME ZONE',
+  TS: 'TIMESTAMP',
+  TZ: 'TIME WITH TIME ZONE',
+  UT: 'UDT Type',
+  XM: 'XML',
+  YM: 'INTERVAL YEAR TO MONTH',
+  YR: 'INTERVAL YEAR',
+  '++': 'TD_ANYTYPE'
+};
+
+const TeradataToAllowedCubeNumbersType: Record<string, string> = {
+  FLOAT: 'numb',
+  DECIMAL: 'dec',
+  INTEGER: 'int',
+  BYTEINT: 'int',
+  SMALLINT: 'int',
+  BIGINT: 'int',
+  NUMBER: 'numb',
+};
+
+export class TeradataDriver extends JDBCDriver {
+  protected readonly config: TeradataDriverConfiguration;
+
+  public static dialectClass() {
+    return TeradataQuery;
+  }
+
+  public constructor(configuration: Partial<TeradataDriverConfiguration>) {
+    const config: TeradataDriverConfiguration = {
+      drivername: 'com.teradata.jdbc.TeraDriver',
+      database: getEnv('teradataDbName', { required: false }),
+      dbType: 'teradata',
+      url: getEnv('teradataUrl'),
+      customClassPath: undefined,
+      properties: {},
+      ...configuration
+    };
+
+    super(config);
+    this.config = config;
+  }
+
+  public readOnly() {
+    return !!this.config.readOnly;
+  }
+
+  protected async getCustomClassPath() {
+    return resolveJDBCDriver();
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async createSchemaIfNotExists(schemaName: string): Promise<any> {
+    // do nothing for the moment.
+  }
+
+  public quoteIdentifier(identifier: string): string {
+    return `"${identifier}"`;
+  }
+
+  private lettersNumbersSpacesDashes(str: string) {
+    return /^[A-Za-z0-9 _]*$/.test(str);
+  }
+
+  public async tableColumnTypes(table: string) {
+    const [databaseName, tableName] = table.split('.');
+    
+    try {
+      let columns = await this.query(`HELP COLUMN * FROM ${tableName}`, []);
+      
+      if (!columns.length) {
+        return [];
+      }
+
+      columns = columns.map((c: any, index) => {
+        const { 'Column Name': columnName, Type: dataType, 'Primary?': keyType } = c;
+        let name = columnName.trim();
+
+        if (!columnName || !dataType) {
+          return null;
+        }
+
+        if (!this.lettersNumbersSpacesDashes(name)) {
+          name = `unknown_${index}_${name}`;
+        }
+
+        const genericType = this.toGenericType(dataType.trim());
+        const columnType = this.toAllowdedNumber(genericType) || genericType;
+        const isPrimaryKey = keyType?.trim() === 'P';
+
+        return { name, type: columnType, attributes: isPrimaryKey ? ['primaryKey'] : [] };
+      }).filter(x => x !== null);
+      return columns;
+    } catch (error) {
+      // TODO: handle errors.
+      console.log(error);
+    }
+    return [];
+  }
+
+  public async getTablesQuery(databaseName: string) {
+    try {
+      // Only SELECT VIEWS (tabkeKind=V)
```

Comment:
@ofir-insait Also, can you provide some insight as to why only Views are supported?

Reply:
The main reason is time, this is what our clients used and what we needed to support.
We have actually started with schema discovery for tables as well and the delta is not huge to add this support.
Teradata obviously has metadata for tables as well.

Note that this is also stated in the commit message:
> Only handling table views
> Handling tables (in addition to table views)
> Currently we only support table views (tableKind=‘V’) and not tables.
> Need to implement and test it.